application/src/test/java/be/jpendel/domain/event/EventFactoryTest.java

```diff
@@ -0,0 +1,24 @@
+package be.jpendel.domain.event;
+
+import be.jpendel.application.CreateEventCommand;
+import org.junit.Test;
+
+import java.time.LocalDateTime;
+
+public class EventFactoryTest {
+
+    @Test
+    public void eventWithSameNameAndSameStartdateLocationCanNotBeCreated(){
+       EventFactory factory = new EventFactory();
+        LocalDateTime time = LocalDateTime.now();
+        CreateEventCommand eventCmd = CreateEventCommand.builder()
+                .withEndDateTime(time).withLocation("loc")
+                .withStartDateTime(time)
+                .withName("test").build();
+        Event e1 = factory.create(eventCmd);
+        EventRepository repo = new InMemEventRepository();
+        repo.add(e1);
+        repo.add(e1);// moet dit falen ?
```

Comment:
En dit lijkt mij dus de betere plek om te falen, maar Guido mag hier zijn 2 cents nog ne keer over geven, de factory is earlier, maar qua responsibility moeilijker.

Alternatief is een repo method findEventWithNameAndDate en dan een dependency van de factory naar de repository.